### PR TITLE
Add basic JAX tutorial

### PR DESCRIPTION
**Pull request by awolant**
## Category:

**New feature**

## Description:

Adds tutorial on how to use DALI with JAX. Tutorial shows how to train simple neural network implemented in JAX with DALI pipelines on MNIST dataset.

### Affected modules and functionalities:

Added tutorial for DALI and JAX. This code builds on an example of MNIST training form JAX codebase. For this reason it was necessary to add JAX licence and attribution to Acknowledgments file. 

### Key points relevant for the review:

- Is the licence and attribution added and correct for the code copied form JAX codebase?
- Is tutorial understandable for the user?

### Tests:

- [ ] Existing tests apply
- [ ] New tests added
  - [ ] Python tests
  - [ ] GTests
  - [ ] Benchmark
  - [ ] Other
- [ ] N/A

### Documentation
- [ ] Existing documentation applies
- [x] Documentation updated
  - [ ] Docstring
  - [ ] Doxygen
  - [ ] RST
  - [x] Jupyter
  - [ ] Other
- [ ] N/A

#### Requirements
- [ ] Implements new requirements
- [ ] Affects existing requirements
- [ ] N/A

**REQ IDs**: N/A

**JIRA TASK**: DALI-3523

### CODE REVIEW

File: docs/examples/frameworks/jax/jax-basic_example.ipynb [hunk 1]
```diff
@@ -0,0 +1,277 @@
+{
```
Comment (klecki) [inline docs/examples/frameworks/jax/jax-basic_example.ipynb:1]: <p>Implementet -&gt; implemented.</p><p>This example is what?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/NVIDIA/DALI/pull/4937/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='0'/>

Comment (klecki) [inline docs/examples/frameworks/jax/jax-basic_example.ipynb:1]: <p>Dot at the end</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/NVIDIA/DALI/pull/4937/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

Comment (klecki) [inline docs/examples/frameworks/jax/jax-basic_example.ipynb:1]: <p>This reads weird for me wrt to our nomenclature. It is usually a pipeline definition function, defining the preprocessing, marked with the decorator etc. After reading "function that will be later used to create instances od DALI pipelines" I thought we are writing some helper for the purpose of JAX. Maybe mention the getting started or similar tutorial that covers the details of writing DALI pipelines?</p><p>You can also list steps of the pipeline in points, reading, decoding etc</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/NVIDIA/DALI/pull/4937/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='3'/>

Comment (klecki) [inline docs/examples/frameworks/jax/jax-basic_example.ipynb:1]: <p>It is already here, cool. Maybe move it to the paragraph that introduces pipeline def?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/NVIDIA/DALI/pull/4937/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='7'/>

Comment (klecki) [inline docs/examples/frameworks/jax/jax-basic_example.ipynb:1]: <p>To get <strong>the</strong> iterator compatible with JAX we need to import <strong>it</strong>  from DALI JAX plugin.</p><p> In addition to the pipeline we can pass the output_map, reader_name and auto_reset <strong>parameters</strong> to the iterator.</p><p> **Here is a quick...**</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/NVIDIA/DALI/pull/4937/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='8'/>

File: docs/examples/frameworks/jax/model.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""A basic MNIST example using Numpy and JAX.
+
+The primary aim here is simplicity and minimal dependencies.
```
Comment (klecki) [inline docs/examples/frameworks/jax/model.py:17]: What is the benefit of putting this in a separate .py file over including those in the tutorial code itself?
Should we at least explain what the utilities do in the tutorial?

If this is ported from JAX, do we have self service SWIPAT for this?

File: docs/examples/frameworks/jax/jax-basic_example.ipynb (hunk 1 above)
Comment (awolant) [reply docs/examples/frameworks/jax/jax-basic_example.ipynb:1]: <p>Done</p>

Comment (awolant) [reply docs/examples/frameworks/jax/jax-basic_example.ipynb:1]: <p>Done. I added <code>pipeline definition</code>	 and more detailed description of what is inside. I moved the links to pipeline tutorials here. </p><p>I left the information that this function creates and instance of the pipeline. I feel that this is necessary to distinguish it from JAX APIs that are generally functional and adding some decorator on a function gives you another functions that you can chain together.</p>

**Review by klecki — APPROVED** [review]

File: docs/examples/frameworks/jax/jax-basic_example.ipynb [hunk 3]
```diff
@@ -0,0 +1,268 @@
...
```
Comment (banasraf) [inline docs/examples/frameworks/jax/jax-basic_example.ipynb:1]: <p>First step is to __ a pipeline definition...</p><p>add? write? implement?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/NVIDIA/DALI/pull/4937/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>

**Review by banasraf — APPROVED** [review]

File: docs/examples/frameworks/jax/jax-basic_example.ipynb (hunk 3 above)
Comment (awolant) [reply docs/examples/frameworks/jax/jax-basic_example.ipynb:1]: <p>Done</p>

**Comment by awolant** [comment]
!build

**Pull request merged**